man/man7/fi_opx.7

```diff
@@ -0,0 +1,218 @@
+
+The above fields have the following meaning:
+
+- \f[I]selector-type\f[]
+The selector criteria the caller opening the context is evaluated against.
+
+- \f[I]hfi-unit\f[]
+The HFI to use if the caller matches the selector.
+
+- \f[I]selector-data\f[]
+Data the caller must match (e.g. NUMA node ID).
+
+Where \f[I]selector-type\f[] is one of the following:
+
+- \f[I]numa\f[]
+True when caller is local to the NUMA node ID given by \f[I]selector-data\f[].
+
+- \f[I]core\f[]
+True when caller is local to the CPU core given by \f[I]selector-data\f[].
+
+And \f[I]selector-data\f[] is one of the following:
+
+- <\f[I]value\f[]> 
+The specific value to match
+
+- <\f[I]range-start\f[]>-<\f[I]range-end\f[]> 
+Matches with any value in that range
+
+In the second form, when opening a context, OPX uses the \f[I]hfi-unit\f[] of the
+first-matching selector. Selectors are evaluated left-to-right. OPX will
+return an error if the caller does not match any selector.
+
+In either form, it is an error if the specified or selected HFI is not in the
+Active state.
+In this case, OPX will return an error and execution will not continue.
+
+With this option, it is possible to cause OPX to try to open more contexts on
+an HFI than there are free contexts on that HFI. In this case, one or more of
+the context-opening calls will fail and OPX will return an error.
+For the second form, as which HFI is selected depends on properties of the caller,
+deterministic HFI selection requires deterministic caller properties. E.g.
+for the \f[I]numa\f[] selector, if the caller can migrate between NUMA domains,
+then HFI selection will not be deterministic.
+
+The logic used will always be the first valid in a selector list. For example, \f[I]default\f[] and
+\f[I]fixed\f[] will match all callers, so if either are in the beginning of a selector list, you will
+only use \f[I]fixed\f[] or \f[I]default\f[] regardles of if there are any more selectors.
+
+Examples:
+
+- \f[I]FI_OPX_HFI_SELECT=1\f[] all callers will open contexts on HFI 0.
+
+- \f[I]FI_OPX_HFI_SELECT=numa:0:0,numa:1:1,numa:0:2,numa:1:3\f[]
+callers local to NUMA nodes 0 and 2 will use HFI 0, callers local to NUMA
+domains 1 and 3 will use HFI 1.
+
+- \f[I]FI_OPX_HFI_SELECT=numa:0:0-3,default\f[] callers local to NUMA nodes 0 thru 3 (including 0 and 3) will use HFI 0, and all else will use default selection logic.
+
+- \f[I]FI_OPX_HFI_SELECT=core:1:0,fixed:0\f[] callers local to CPU core 0 will use HFI 1, and all others will use HFI 0.
+
+- \f[I]FI_OPX_HFI_SELECT=default,core:1:0\f[] all callers will use default HFI selection logic.
+
+
+.SH RUNTIME PARAMETERS
+.PP
+Environment variables starting with FI are Libfabric related.
+.br
+OPX provider requires the following  environment variables:
+.TP
+.B \f[I]FI_PROVIDER\f[R]
+This can be set using FI_PROVIDER=opx.
+.TP
+
+.SH SEE ALSO
+.PP
+\f[C]fabric\f[](7), \f[C]fi_provider\f[](7), \f[C]fi_getinfo\f[](3)
+.SH AUTHORS
+Cornelis Networks.
```

Comment:
Btw - the entire contents of this file will be replaced by a scripts that will process the markdown file. You only need to maintain the .md file.

Reply:
ok. We weren't sure due to them both being in git.